util.py

```diff
@@ -260,14 +260,22 @@ def create_executable(filename, command):
     )
 
 
-def check_file_sha256_checksum(filename, reference_checksum):
+def compute_file_sha256_checksum(filename):
     """
     Computes the sha256sum for a file
 
     :param str filename: a single file to be checked
-    :param str reference_checksum: reference sha256 checksum
-    :return:
+    :return: checksum
+    :rtype:str
     """
     checksum_command_output = sp.check_output(["sha256sum", filename])
-    file_checksum = checksum_command_output.decode().strip().split(" ")[0]
-    assert file_checksum == reference_checksum
+    return checksum_command_output.decode().strip().split(" ")[0]
+
+
+def check_file_sha256_checksum(filename, reference_checksum):
```

Comment:
For the use cases you have above, you could also add a small function which directly takes two files as input and computes and compares their checksums.

Reply:
Are you sure that testing the checksum is the best way to perform the test? This way, it also takes into account the output formatting and not only the content of the data structures

Reply:
I think this depends on the kind of job. If you are checking the data structures the tests will be more complicated, and then people will not add tests. This is why I thought a checksum is the easiest solution for a start.